src/test/java/org/kainos/ea/service/JobRoleServiceTest.java

```diff
@@ -101,4 +110,30 @@ public void updateJobRole_shouldReturnID_whenSuccess() throws SQLException, Fail
         Mockito.when(jobRoleDao.updateJobRole(any(int.class), any(UpdateJobRoleRequest.class))).thenReturn(1);
         assertEquals(1, jobRoleService.updateJobRole((short) 1, jobRoleRequest));
     }
+
+    @Test
+    public void createJobRole_shouldReturnID__whenDaoReturnsId() throws SQLException, FailedToCreateJobRoleRequestException, InvalidJobRoleException {
+        int expectedResult = 1;
+        Mockito.when(createJobRoleValidator.isValidJobRole(any(CreateJobRoleRequest.class))).thenReturn(null);
+        Mockito.when(jobRoleDao.createJobRole(createJobRoleRequest)).thenReturn(expectedResult);
+
+        int result = jobRoleService.createJobRole(createJobRoleRequest);
+        assertEquals(result, expectedResult);
+    }
+
+    @Test
+    public void createJobRole_shouldThrowFailedToCreateJobRoleRequestException_whenDaoThrowsSQLException() throws SQLException, FailedToCreateJobRoleRequestException {
+
+        Mockito.when(createJobRoleValidator.isValidJobRole(any(CreateJobRoleRequest.class))).thenReturn(null);
+        Mockito.when(jobRoleDao.createJobRole(createJobRoleRequest)).thenThrow(FailedToCreateJobRoleRequestException.class);
+
+        assertThrows(FailedToCreateJobRoleRequestException.class, () -> jobRoleService.createJobRole(createJobRoleRequest));
+    }
+
+    @Test
+    public void createJobRole_shouldThrowInvalidJobRoleException_whenInvalidJobRoleRequest() throws InvalidJobRoleException {
+        Mockito.when(createJobRoleValidator.isValidJobRole(any(CreateJobRoleRequest.class))).thenReturn("Test invalid job role");
+        assertThrows(InvalidJobRoleException.class,
+                () -> jobRoleService.createJobRole(createJobRoleRequest));
+    }
 }
```

Comment:
Missing the test case where DAO returns -1

Reply:
added